packages/webapp/pages/history.tsx

```diff
@@ -50,24 +51,12 @@ const History = (): ReactElement => {
   const { data, isInitialLoading, isLoading, hasData, infiniteScrollRef } =
     useInfiniteReadingHistory({ ...queryProps });
 
-  useEffect(() => {
-    if (tokenRefreshed && !user) {
-      router.replace('/');
-    }
-  }, [tokenRefreshed, user]);
-
-  if (!hasData && !isLoading) {
-    return (
-      <>
-        {seo}
-        <ReadingHistoryEmptyScreen />
-      </>
-    );
-  }
-
   return (
-    <>
-      {seo}
+    <ProtectedPage
```

Comment:
Just wondering if the layouts won't be a better option for this?

Reply:
Are you referring to the function `getLayout`? Or the NextJS layout?

Reply:
Basically the Next layouts indeed, we could opt to create a ProcetedPage layout that could wrap around everything?
But open to this one as well so leaving it up to you.

Reply:
Ah, interesting. Will look into that, might be something quick we can implement, at least we can start now.

Reply:
We are already utilizing the `getLayout` property, the current implementation should be fine with its own layout as the `getLayout` will provide the nesting layout.